### PR TITLE
Fcrepo 849

### PR DESCRIPTION
**Pull request by barmintor**
I think this will fix a RELS-INT problem we're having, and may facilitate some migrations to 4 that involve moving from inline to managed content en route

### CODE REVIEW

File: fcrepo-server/src/main/java/org/fcrepo/server/resourceIndex/ResourceIndex.java [hunk 1]
```diff
@@ -79,6 +86,18 @@ void modifyObject(DOReader oldReader, DOReader newReader)
...
+     * Export the triples indexed from a given object.
+     * Adding an object is the equivalent of inserting these triples.
+     * Deleting an object is the equivalent of deleting these triples.
+     * Modifying an object is the equivalent of processing the difference
+     * between the export before and after a change.
+     * @param object
+     * @return
+     * @throws ResourceIndexException
+     */
+    List<Triple> exportObject(DOReader object) throws ResourceIndexException;
```
Comment (ajs6f) [inline fcrepo-server/src/main/java/org/fcrepo/server/resourceIndex/ResourceIndex.java]: How come this is an (ordered) list?

File: fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultDOManager.java [hunk 2]
```diff
@@ -1117,7 +1120,11 @@ public void doCommit(boolean cachedObjectRequired, Context context,
                     null, null, null, obj));
 
             try { // for cleanup catch
-
+                boolean riEnabled = (m_resourceIndex != null &&
+                        m_resourceIndex.getIndexLevel() != ResourceIndex.INDEX_LEVEL_OFF);
+                @SuppressWarnings("unchecked")
+                List<Triple> before = (riEnabled && !obj.isNew()) ? m_resourceIndex.exportObject(
+                        getReader(false, null, obj.getPid())) : Collections.EMPTY_LIST;
```
Comment (ajs6f) [inline fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultDOManager.java]: Isn't it better `Collections.emptyList()`?

File: fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultDOManager.java [hunk 3]
```diff
@@ -1336,14 +1343,25 @@ public void doCommit(boolean cachedObjectRequired, Context context,
                 if (m_resourceIndex != null &&
                         m_resourceIndex.getIndexLevel() != ResourceIndex.INDEX_LEVEL_OFF) {
                     logger.info("Adding to ResourceIndex");
+                    @SuppressWarnings("unchecked")
+                    List<Triple> after = (riEnabled) ? m_resourceIndex.exportObject(
+                            new SimpleDOReader(null,
+                                    null, null, null, null, obj)) : Collections.EMPTY_LIST;
```
Comment (ajs6f) [inline fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultDOManager.java]: Ditto.

File: fcrepo-server/src/main/java/org/fcrepo/server/resourceIndex/ResourceIndex.java (hunk 1 above)
Comment (barmintor) [inline fcrepo-server/src/main/java/org/fcrepo/server/resourceIndex/ResourceIndex.java]: It lines up better with the org.trippi.TriplestoreWriter interface that this interface extends.

File: fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultDOManager.java (hunk 2 above)
Comment (barmintor) [inline fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultDOManager.java]: Yes, except that the factory method has a type, and as such can't be cast to a typed List.

File: fcrepo-server/src/main/java/org/fcrepo/server/resourceIndex/ResourceIndex.java (hunk 1 above)
Comment (ajs6f) [inline fcrepo-server/src/main/java/org/fcrepo/server/resourceIndex/ResourceIndex.java]: I love you you, Trippi. I love you so much.

File: fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultDOManager.java (hunk 2 above)
Comment (ajs6f) [inline fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultDOManager.java]: Eh? It's generic, no? I put Collections.emptyList() in typed lists all the time...

Comment (barmintor) [inline fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultDOManager.java]: Well then why does Eclipse yell at me?

Comment (ajs6f) [inline fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultDOManager.java]: Dunno. When this finished building (love them F3 build times!) I'll import it and tinker.

**Pull request merged**